### PR TITLE
Complete API Provider creation flow

### LINKED ISSUE #3476: Update "save or add step" page to implement "add flow/integration" page based on design
Update [this](https://github.com/syndesisio/syndesis/tree/master/app/ui/src/app/integration/edit-page/save-or-add-step) page to suit [this](https://redhat.invisionapp.com/share/3CNPQLOKAEW#/screens/316356339) design.  This page in general is where you wind up after completing a sequence of pages in the editor.

### LINKED ISSUE #3477: Implement "configure return path" page based on designs
[This](https://redhat.invisionapp.com/share/3CNPQLOKAEW#/screens/316356340) design, think it'll be a new page.  This is just for the page, #3487 is more related to adjusting the editor's current work flow to handle having the finish connection instead be the return path indicator.

### LINKED ISSUE #3478: Update IVP to show the operation/method/path when editing a flow per the design

### LINKED ISSUE #3479: Implement API provider santa hat for IVP per the design

### LINKED ISSUE #3482: Update integration detail page to add support for API provider multi flow information
Design -> https://redhat.invisionapp.com/share/4JNR7J5DMNY#/screens/316053627

### PR DESCRIPTION
**Pull request by riccardo-forina**
This implements the missing steps in the API Provider creation flow:

* it's now mandatory to specify an integration name before editing the Operations
* the sidebar is API Provider aware, it shows the right UI elements and disables the interaction with the API Provider connector icon

Closes #3476 
Closes #3477
Closes #3478 
Closes #3479 
Closes #3482

## Todo

* [x] Text in the "Save integration" step are placeholders. @TovaCohen could you help with this? [This](https://user-images.githubusercontent.com/966316/46282403-b1d06100-c571-11e8-9191-c9889150eca2.png) is the screenshot of the step I'm talking about. In short, to be able to edit the Operations read from an OpenAPI specification file we first need to save the integration.

## Screenshots

api-provider-creation-demo

### CODE REVIEW

File: app/ui/src/app/integration/edit-page/flow-view/flow-view.component.html [hunk 1]
```diff
@@ -76,37 +76,46 @@
...
+            <syndesis-integration-flow-view-step [step]="endConnection()"
+                                                  [position]="lastPosition()"
                                                   [currentPosition]="currentPosition"
                                                   [currentState]="currentState"></syndesis-integration-flow-view-step>
           </div>
+        </ng-container>
 
-          <!-- Middle Steps: Save or Add Step -->
-          <ng-container *ngTemplateOutlet="insertStepTemplate; context: { position: (position + 1) }"></ng-container>
+        <ng-template #apiProviderSanta>
```
Comment (gashcrumb) [inline app/ui/src/app/integration/edit-page/flow-view/flow-view.component.html]: lol

Comment (riccardo-forina) [reply app/ui/src/app/integration/edit-page/flow-view/flow-view.component.html]: Ooops! 😂 Let me pick a better name...

**Comment by TovaCohen** [comment]
@riccardo-forina - How's this:

Give this integration a name

To add OpenAPI operations to this integration, specify a name for this integration so Syndesis/Fuse Online can save your work in a draft version of the integration.

**Comment by heiko-braun** [comment]
Good work @riccardo-forina !

**Pull request merged**